docs/api/csf/csf-next.mdx

```diff
@@ -0,0 +1,422 @@
+    // ...current config
+    // 👇 Add your addons here
++   addons: [addonA11y()],
++ });
+- };
+- export default preview;
+```
+
+**3. Update your story files**
+
+Story files have been updated for improved usability. With the new format:
+
+- Import the preview construct from the Storybook preview file
+- The meta object is now created via the [`preview.meta`](#previewmeta) function and does not have to be exported as a default export
+- Stories are now created from the meta object, via the [`meta.story`](#metastory) function
+
+<Callout variant="info">
+The examples below show the changes needed to upgrade a story file from CSF 3 to CSF Next. You can also upgrade from CSF 1 or 2 using similar steps.
+</Callout>
+
+```diff title="Button.stories.js|ts"
++ import preview from '../.storybook/preview';
+- import type { Meta, StoryObj } from '@storybook/your-framework';
+
+import { Button } from './Button';
+
++ const meta = preview.meta({
+- const meta = {
+    // ...meta object is unchanged
++ });
+- } satisfies Meta<typeof Button>;
+- export default meta;
+
+- type Story = StoryObj<typeof meta>;
+
++ export const Primary = meta.story({
+- export const Primary: Story = {
+    // ...story object is unchanged
++ });
+- };
+```
+
+Note that importing or manually applying any type to the meta or stories is no longer necessary. Thanks to the factory function pattern, the types are now inferred automatically.
+
+**3.1 Reusing story properties**
+
+<Callout variant="info" icon="💡">
+
+If you are reusing story properties to create a new story based on another, the [`<StoryName>.extend`](#storynameextend) method is the recommended way to do so.
+
+</Callout>
+
+Previously, story properties such as `Story.args` or `Story.parameters` were accessed directly when reusing them in another story. While accessing them like this is still supported, it is deprecated in CSF Next.
+
+All of the story properties are now contained within a new property called `composed` and should be accessed from that property instead. For instance, `Story.composed.args` or `Story.composed.parameters`.
+
+```diff title="Button.stories.js|ts"
+// ...rest of file
+
++ export const Primary = meta.story({
+- export const Primary: Story = {
+    args: { primary: true },
++ });
+- };
+
++ export const PrimaryDisabled = meta.story({
+- export const PrimaryDisabled: Story = {
+    args: {
++     ...Primary.composed.args,
+-     ...Primary.args,
+      disabled: true,
+    }
++ });
+- };
+```
+
+<Callout variant="info">
+  The property name "composed" was chosen because the values within are composed from the story, its component meta, and the preview configuration.
+
+  If you want to access the direct input to the story, you can use `Story.input` instead of `Story.composed`.
+</Callout>
+
+**4. Update your Vitest setup file**
+
+If you're using [Storybook's Vitest addon](../../writing-tests/integrations/vitest-addon.mdx), you can remove your Vitest setup file.
```

Comment:
They might have other setup stuff in there that they don't want to remove. I guess users will figure this out on their own, but perhaps there is a way to phrase it better.

Reply:
I clarified which setup file we're talking about. It seems very unlikely they're using the Storybook-specific setup file for anything else?

```diff
+ ...you can remove your Vitest setup file (`.storybook/vitest.setup.ts`).
- ...you can remove your Vitest setup file.
```